### DISCUSSION TITLE
Extracting analysis status

### DISCUSSION THREAD

**Discussion by linokl** [JWock82/Pynite#142] [General]
Hi Graig,

Happy new year and hope you are doing well. I wanted to give you an update on the standardsolver. I mange to create a frame3d analysis module for the standard solver. Few screenshots below and also link to a reveal YouTube video I did. 
![Screenshot 2022-12-31 at 19 21 40](https://user-images.githubusercontent.com/20646443/210457939-f2360bfe-9648-4d03-bc1a-a7949d2aa266.png)
![Screenshot 2022-12-31 at 19 18 19](https://user-images.githubusercontent.com/20646443/210457958-f47a4fbe-564e-4972-972d-35be8807df61.png)

YouTube video link below.
https://youtu.be/Ihyax0vRwck

Is there a way once you have run an analysis on a model to extract why the analysis failed, if it happen to fail for example due to matrix being singular or instability? It is printed in the console but I am not able to get it in to a python variable and pass it to the user in the user interface so user may get a better understanding why the model is failed. Is there a work around for it?

Thank you

Regards
Lino

**Comment by JWock82** [comment]
Hi Lino. It's been a while since I programmed this. I believe PyNite throws an exception for each case. I think you could use a try/catch block to try running the model and to catch the exception's details if it doesn't run.

By the way, a singular stiffness matrix is still an unstable model. It represents a global model instability, rather than just a local nodal instability.

Thanks,

Craig Brinck
________________________________
From: linokl ***@***.***>
Sent: Tuesday, January 3, 2023 4:30 PM
To: JWock82/PyNite ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [JWock82/PyNite] Extracting analysis status (Discussion #142)


Hi Graig,

Happy new year and hope you are doing well. I wanted to give you an update on the standardsolver. I mange to create a frame3d analysis module for the standard solver. Few screenshots below and also link to a reveal YouTube video I did.
[Screenshot 2022-12-31 at 19 21 40]<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F20646443%2F210457939-f2360bfe-9648-4d03-bc1a-a7949d2aa266.png&data=05%7C01%7C%7C32b6cb6e92fe4db7f59e08daede27252%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638083854083790161%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=uSCOAYs46RNBsVKG9IblNznPN8zxXdMoogGAOoA5sOY%3D&reserved=0>
[Screenshot 2022-12-31 at 19 18 19]<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F20646443%2F210457958-f47a4fbe-564e-4972-972d-35be8807df61.png&data=05%7C01%7C%7C32b6cb6e92fe4db7f59e08daede27252%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638083854083790161%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6DA2iRFUdgBiimdhVbSHc0JvBSE8M7XNwi3tyStAvxI%3D&reserved=0>

YouTube video link below.
https://youtu.be/Ihyax0vRwck<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fyoutu.be%2FIhyax0vRwck&data=05%7C01%7C%7C32b6cb6e92fe4db7f59e08daede27252%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638083854083790161%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Qf7hc6SWzaHqI76Hpn%2Ft8DirPYGvOqI1fXAZFvtC0XA%3D&reserved=0>

Is there a way once you have run an analysis on a model to extract why the analysis failed, if it happen to fail for example due to matrix being singular or instability? It is printed in the console but I am not able to get it in to a python variable and pass it to the user in the user interface so user may get a better understanding why the model is failed. Is there a work around for it?

Thank you

Regards
Lino

**Reply by linokl** [reply]
Hi Craig,

Thanks for the reply. I will try that. I will send you link to the frame3d module soon your feed back will be greatly appreciated. I am planning to build Frame3D as a paid premium feature after a while may be free for an year then a paid module, if there is users for it so that it will cover the server and compute cost..if I manage get some users and make any revenue out of it, hopefully I can contribute towards supporting your work in some way as well :)

Regards
Lino

**Comment by linokl** [comment]
The 3d module alpha is now online :)

link: https://frame3d.standardsolver.com 

I have removed the rotational spring restraints due to the bug. Please let me know if you have any comments.

Regards
Lino
